### PR TITLE
Add LightningLite to top level imports

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Makes it possible to import LightningLite directly from the main lightning package, just like Trainer, LightningWork, etc.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

**Review by lantiga — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by otaj — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: .github/workflows/ci-pkg-install.yml [hunk 1]
```diff
@@ -63,6 +63,8 @@ jobs:
         run: python -m lightning --version
 
       - name: DocTest package
+        env:
+          PY_IGNORE_IMPORTMISMATCH: 1  # needed for Python 3.7
```
Comment (awaelchli) [inline .github/workflows/ci-pkg-install.yml:67]: @carmocca @Borda I had to add this here. I couldn't find another solution. The errors I was getting:

https://github.com/Lightning-AI/lightning/actions/runs/3421692912/jobs/5698175906

```
==================================== ERRORS ====================================
_____________ ERROR collecting src/lightning/pytorch/__about__.py ______________
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/_pytest/runner.py:339: in from_call
    result: Optional[TResult] = func()
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/_pytest/runner.py:370: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/_pytest/doctest.py:551: in collect
    mode=self.config.getoption("importmode"),
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/_pytest/pathlib.py:556: in import_path
    raise ImportPathMismatchError(module_name, module_file, path)
E   _pytest.pathlib.ImportPathMismatchError: ('lightning.pytorch.__about__', '/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/lightning/pytorch/__about__.py', PosixPath('/home/runner/work/lightning/lightning/src/lightning/pytorch/__about__.py'))


...
```

Is it ok with you?

Comment (carmocca) [reply .github/workflows/ci-pkg-install.yml:67]: I don't know what the error is trying to say, but if you want to ignore it, then I'd suggest

```suggestion
        env:
          PY_IGNORE_IMPORTMISMATCH: ${{ matrix.python-version == "3.7" }}
```

Comment (awaelchli) [reply .github/workflows/ci-pkg-install.yml:67]: Sounds good!

**Review by akihironitta — APPROVED** [review]

**Pull request merged**